services/builder/configProcessor.go

```diff
@@ -131,7 +145,7 @@ func stateDifference(config *pb.Config) (*pb.Config, map[string]*nodepoolsCounts
 
 	//if any key left, it means that nodepool is defined in current state but not in the desired, i.e. whole nodepool should be deleted
 	if len(currentNodepoolMap) > 0 {
-		log.Info().Msgf("Detected deletion of a nodepools")
+		//log.Info().Msgf("Detected deletion of a nodepools") NOTE:DEBUG
```

Comment:
I think these changes are great and add more context to the logs.

Could we instead of `//log.Info().Msgf("Detected deletion of a nodepools") NOTE:DEBUG` just use `log.Debug().Msgf(...)` and have the log level take from an ENV variable ? In the CI pipeline it could run in DEBUG but the default could be just INFO,

Reply:
No problem, just thinking out loud here. Both solutions are ok, though I would lean towards the latter as maintaining two files is just more work.

Reply:
Changes implemented, please check if this is what you had in mind

Reply:
Yes that's what I had in mind 👍